File: backend/src/main/java/com/coing/domain/coin/market/dto/MarketDto.java
package com.coing.domain.coin.market.dto;

import com.coing.domain.coin.market.entity.Market;
import com.fasterxml.jackson.annotation.JsonProperty;

public record MarketDto(
	@JsonProperty("market")
	String market,
	@JsonProperty("korean_name")
	String koreanName,
	@JsonProperty("english_name")
	String englishName
) {
	public Market toEntity() {
		return Market.builder()
			.code(market)
			.koreanName(koreanName)
			.englishName(englishName)
			.build();
	}
}

File: backend/src/main/java/com/coing/domain/coin/market/dto/MarketDto.kt
package com.coing.domain.coin.market.dto

import com.coing.domain.coin.market.entity.Market
import com.fasterxml.jackson.annotation.JsonProperty

data class MarketDto(
	@JsonProperty("market")
	val market: String,

	@JsonProperty("korean_name")
	val koreanName: String,

	@JsonProperty("english_name")
	val englishName: String
) {
	fun toEntity(): Market {
		return Market(
			code = market,
			koreanName = koreanName,
			englishName = englishName
		)
	}
}